lib/sandbox_results_handler.rb

```diff
@@ -24,17 +24,20 @@ def generate_data_for_frontend(test_output, package_type)
 
   def test_results(test_output)
     # Push test results into exercise's error messages
-    return if test_output['testResults'].empty? || test_output['testResults'].first['successful']
+    return if test_output['testResults'].empty? || test_output['status'] == 'PASSED'
     header = 'Virheet testeissä: '
-    messages = test_output['testResults'][0]['message']
+    messages = ''
+    test_output['testResults'].each do |test|
+      messages += "#{test['message']}<rivinvaihto>"
+    end
     error = { header: header, messages: messages }
     @exercise.error_messages.push error
   end
 
   def compile_errors(test_output, package_type)
     return unless test_output['status'] == 'COMPILE_FAILED'
     header = package_type == 'TEMPLATE' ? 'Tehtäväpohja ei kääntynyt: ' : 'Malliratkaisu ei kääntynyt: '
-    messages = error_message_lines(test_output).join('\n')
+    messages = error_message_lines(test_output).join('<rivinvaihto>')
```

Comment:
:finland: ?

Reply:
Why add rivinvaihto here?

Reply:
'rivinvaihto' so that crowdsorcerer won't get all confused.